src/main/java/de/doubleslash/keeptime/view/ProjectsListViewController.java

```diff
@@ -69,18 +70,23 @@ public ProjectsListViewController(final Model model, final Controller controller
       this.hideable = hideable;
       this.mainStage = mainStage;
       this.availableProjectsListView = availableProjectsListView;
-      availableProjectsListView.setCellFactory(callback -> returnListCellOfProject());
+      availableProjectsListView.setCellFactory(listView -> returnListCellOfProject());
 
       filteredData = new FilteredList<>(model.getSortedAvailableProjects(), p -> true);
       availableProjectsListView.setItems(filteredData);
 
       projectSelectionNodeMap = new HashMap<>(model.getAvailableProjects().size());
 
-      for (final Project project : model.getSortedAvailableProjects()) {
-         if (project.isEnabled()) {
-            final Node node = addProjectToProjectList(project);
-            projectSelectionNodeMap.put(project, node);
+      model.getSortedAvailableProjects().addListener((ListChangeListener<? super Project>) listener -> {
+         listener.next();
+         if (listener.wasAdded()) {
+            final Project addedProject = listener.getAddedSubList().get(0);
+            addProjectToProjectSelectionNodeMap(addedProject);
          }
+      });
```

Comment:
Put the logic into the updateItem method. If it finds an entry in the nodemap use it, otherwise create it. 
Like it is now, it is quite complex and dependent of each other (this listener has to be exectued before the updateItem - which we cannot guarantee)